### PR TITLE
feat(pipelineloop): Support last_idx and last_elem

### LINKED ISSUE #1041: Introduce new pipelineloop variable "last_idx" and "last_elem" for sequential loop.
/kind feature

**Description:**
Introduce new pipelineloop variable "last_idx" and "last_elem" for sequential loop. The "last_idx" and "last_elem" are pipelineloop output parameters that can be used in another task. The implementation will work as the follow:

1. Introduce new output parameters "last_idx" and "last_elem" for all loop. 
2. In the pipelineloop controller, have an in memory key value store that points to the last active loop index and element. 
3. Once the pipelineloop finish/break, output "last_idx" and "last_elem" with the in memory values.
4. The Python SDK will need to map loop.last_idx and loop.last_elem to these two new output parameters.

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by ScrapCodes**
**Which issue is resolved by this Pull Request:** 
Resolves #1041

**Description of your changes:**

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

cc @Tomcli

### CODE REVIEW

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go [hunk 1]
```diff
@@ -714,7 +749,7 @@ func (c *Reconciler) updatePipelineRunStatus(ctx context.Context, iterationEleme
 	if err != nil {
 		return 0, nil, nil, fmt.Errorf("could not list PipelineRuns %#v", err)
 	}
-	if pipelineRuns == nil || len(pipelineRuns) == 0 {
+	if len(pipelineRuns) == 0 {
```
Comment (Tomcli) [inline tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:752]: What error will you get for len(pipelineRuns) if pipelineRuns is nil? should you also handle this case?

Comment (ScrapCodes) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:752]: As per this, https://go.dev/tour/moretypes/12 len() on Nil slices in golang return 0. So additional check of slice == nil was redundant.

**Comment by Tomcli** [comment]
thanks @ScrapCodes, the pipeline loop code looks good. would you mind to rebase this to trigger the end to end test? thanks.

**Comment by Tomcli** [comment]
@ScrapCodes if you think you need more time for the sdk, maybe you can revert the sdk changes for now and we can merge this PR with just the pipeline loop code. This way we can better identify problems when we need to revert certain commit.

**Comment by Tomcli** [comment]
thanks @ScrapCodes 

/lgtm
/approve

**Pull request merged**